### ISSUE TITLE
Some strange SSL issue between Wi-Fi and 4G

### ISSUE THREAD

**Issue by GregOriol** [AirenSoft/OvenMediaEngine#629]
**PROBLEM**

Hi,

We are seing a very strange behavior with OME v0.12.9 around SSL connexions, that may or may not be linked to SNI, but all this is not really clear to us so far.
The problem is: with the same browser (Safari on iOS 15, Chrome on Android), when connected through normal "home" Wi-Fi, the player can play the webrtc stream, but when connected through 4G, the player can't play it and there are these errors in the ome logs:
```
[2022-01-10 18:52:45.308] W [SPRtcSig-T443:38279] HttpServer | https_server.cpp:170  | Could not find certificate for server_name: 123.123.123.123
[2022-01-10 18:52:45.308] W [SPRtcSig-T443:38279] OpenSSL | tls_context.cpp:229  | Could not select certificate: 123.123.123.123
```
(where 123.123.123.123 is the IP address of the server)

The Server.xml contains this SSL configuration:
```
<Host>
    <Names>
        <Name>ome.mydomain.com</Name>
    </Names>
    <TLS>
        <CertPath>/etc/letsencrypt/live/ome.mydomain.com/cert.pem</CertPath>
        <KeyPath>/etc/letsencrypt/live/ome.mydomain.com/privkey.pem</KeyPath>
        <ChainCertPath>/etc/letsencrypt/live/ome.mydomain.com/fullchain.pem</ChainCertPath>
    </TLS>
</Host>
```

We tried adding `<Name>123.123.123.123</Name>` which makes the log disappear, but the player still can't play the webrtc stream.

We don't really know what to look for here: could the 4G network somehow provide different SNI information that OME can't understand? There doesn't seem to be such an issue with any other servers we have (webrtc with jitsi, nginx, ...).

Does anyone have any idea on how to investigate? what to look for?

What is really strange is that jitsi, in the same project, works as expected on webrtc with the same SSL certificate.

**ENVIRONMENTAL INFORMATION**

- OvenMediaEngine Information

Release v0.12.10 doesn't log anything when the player tries to connect, with or without the `<Name>ip</Name>`
Release v0.12.9 as described above
Release v0.12.8 before SNI was added too, this version doesn't log anything when the player tried to connect though

- Encoder information

OBS

- Player information

Player v0.10.15
Reproduced on the demo player
The console shows an SSL error, here in Safari on iOS 15:
<img width="1876" alt="Capture d’écran 2022-01-10 à 20 46 16" src="https://user-images.githubusercontent.com/3975044/148829481-20d68891-3887-4910-884d-d78c4c834a29.png">

- URL

Can't provide a working URL

- Operating system and version

Ubuntu 20.04

**SETUP INFORMATION AND LOGS**

[Server.xml.txt](https://github.com/AirenSoft/OvenMediaEngine/files/7841774/Server.xml.txt)

[ovenmediaengine.log](https://github.com/AirenSoft/OvenMediaEngine/files/7841807/ovenmediaengine.log)

**OTHER HELPFUL INFORMATION**

**Comment by GregOriol** [comment]
Not sure yet, but that may actually be related to https://github.com/AirenSoft/OvenMediaEngine/issues/112 as the Wi-Fi seems to use IPv4 and the 4G seems to use IPv6

**Comment by getroot** [comment]
Thanks for the detailed report.

First of all, the log below occurs because you accessed the wss url by IP instead of domain. wss must access the domain applied to the configured certificate.

```
[2022-01-10 18:52:45.308] W [SPRtcSig-T443:38279] HttpServer | https_server.cpp:170 | Could not find certificate for server_name: 123.123.123.123
```

I rather find it strange that "wss://ip" works on wifi.

Would you like to try playing with wss://domain first? And please share any additional information you find out.

And 0.12.9 and 0.12.10 make little difference. This is just one emergency patch applied to ICE, and the signaling and certificate behavior are exactly the same. If the behavior of these two versions is different, something is wrong with the test.

I will also look into this matter further.

**Comment by GregOriol** [comment]
Sorry for the messiness of the report though, it was a live investigation and there may be different things in it, still not entirely sure.

At least for the IP access, I'm sure that we did not try to call wss://ip, only wss://domain (our code and demo player url was the same for all those tests) so it's still unclear why the certificate was being looked up for the ip and not the domain.

As for the main issue connectivity issue, it is now clear that was 4G trying to access the stream with IPv6, which failed.

**Comment by getroot** [comment]
I'm closing this issue because I believe it can be resolved with #112. If this problem has other causes, please open it again.
